### PR TITLE
API-8973 availability report

### PR DESCRIPTION
**Pull request by MaciejSzaflik**
# 🚀 Links

- [Feature branch](https://developers.labs.livechat.com/docs/feature/API-8973-availablity-report)
- [Jira](https://livechatinc.atlassian.net/browse/API-8969)

# 📓 Description
[Docs from old api](https://github.com/livechat/livechat-docs/blob/master/livechat2/rest-api/docs/chatReports.md#get-availability-report)
This report have different filters and endpoint so I added new table under main one - filters behave in the same way as this above, we just have less of them (the same as in old api)

### CODE REVIEW

File: src/pages/data-reporting/changelog/index.mdx [hunk 1]
```diff
@@ -65,6 +65,10 @@ The developer preview version provides a preview of the upcoming changes to the
 - There's a new method, [**Ratings**](/data-reporting/reports-api/v3.4/#ratings), which shows the number of rated chats along with their ratings during a specified period of time.
 - There's a new method, [**Ranking**](/data-reporting/reports-api/v3.4/#ranking), which shows the ratio of good to bad ratings for each operator agent within a license.
 
+### Agents
+
+- There's a new method, [**Availability**](/data-reporting/reports-api/v3.4/#availability), which shows how much the agent (or group or whole account) was available for chatting during specified period of time.
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/changelog/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 2]
```diff
@@ -414,6 +430,62 @@ https://api.livechatinc.com/v3.4/reports/chats/ranking \
...
 </Section>
 
+## Agents
+
+<Section>
+<Text>
+
+### Availability
+
+Shows how much the agent (or group or whole account) was available for chatting during specified period of time.
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 3]
```diff
@@ -414,6 +430,62 @@ https://api.livechatinc.com/v3.4/reports/chats/ranking \
...
+| `records.minutes`        | Total minutes agent was available that `hour` (Only for `hour` distribution) |
+
+</Text>
+<Code>
+<CodeSample path={'REQUEST'}>
+<Sample>
+
+```shell
+curl -X POST \
+https://api.livechatinc.com/v3.4/reports//availability \
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 4]
```diff
@@ -71,6 +74,19 @@ To call the Reports API, you need to use a Bearer access token. [Learn how to ge
...
+| `filters.from`                                                        | No       | `string`  | Date & time format compatible with RFC3339 with optional resolution of microseconds, `YYYY-MM-DDTHH:MM:SS.ssssssZHH:MM`.                                                                                                                    |
+| `filters.to`                                                          | No       | `string`  | Date & time format compatible with RFC3339 with optional resolution of microseconds, `YYYY-MM-DDTHH:MM:SS.ssssssZHH:MM`.                                                                                                                    |
+| `filters.agents.<string_filter_type>`<sup>**2**</sup>                 | No       | `any`     | Described below; `exists` set to `false` will return unassigned chats; `true` will return the assigned ones.                                                                                                                                |
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 5]
```diff
@@ -414,6 +430,62 @@ https://api.livechatinc.com/v3.4/reports/chats/ranking \
...
+
+|                     |                                                                    |
+| ------------------- | -------------------------------------------------------------------|
+| **Method URL**      | `https://api.livechatinc.com/v3.4/reports/agents/availability`     |
+| **HTTP method**     | POST, GET                                                          |
+| **Required scopes** | -                                                                  |
+
+#### Request
+
+See [available parameters and filters](#available-parameters-and-filters). Depending on distribution parameter response will contain diffrent time resolution.
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 6]
```diff
@@ -414,6 +430,62 @@ https://api.livechatinc.com/v3.4/reports/chats/ranking \
...
+
+#### Response
+
+| Field                    | Notes                                                                        |
+| ------------------------ | -----------------------------------------------------------------------------|
+| `total`                  | The total time agent was available in the specified date range.              |
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 7]
```diff
@@ -414,6 +430,62 @@ https://api.livechatinc.com/v3.4/reports/chats/ranking \
...
+| `records`                | Contains the `distribution` objects, for example, `day`.                     |
+| `records.hours`          | Total hours agent was available that `day` (Only for `day` distribution)     |
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 8]
```diff
@@ -414,6 +430,62 @@ https://api.livechatinc.com/v3.4/reports/chats/ranking \
...
+| `records.minutes`        | Total minutes agent was available that `hour` (Only for `hour` distribution) |
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 9]
```diff
@@ -49,12 +49,15 @@ To call the Reports API, you need to use a Bearer access token. [Learn how to ge
...
-| --------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
-| **Chats** | [`agents_chatting_duration`](#agents-chatting-duration) [`tags`](#tags) [`total_chats`](#total-chats) [`ratings`](#ratings) [`ranking`](#ranking) |
+|            |                                                                                                                                                   |
+| ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **Chats**  | [`agents_chatting_duration`](#agents-chatting-duration) [`tags`](#tags) [`total_chats`](#total-chats) [`ratings`](#ratings) [`ranking`](#ranking) |
+| **Agents** | [`availability`](#availability)                                                                                                                   |
 
 ## Available parameters and filters
 
+### For **Chats** reports
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 10]
```diff
@@ -71,6 +74,19 @@ To call the Reports API, you need to use a Bearer access token. [Learn how to ge
 | `filters.event_types.<event_type_filter_type>`<sup>**5**</sup>        | No       | `any`     | Described below.                                                                                                                                                                                                                            |
 | `filters.groups.values`                                               | No       | `array`   | Array of group IDs.                                                                                                                                                                                                                         |
 
+### For **Agents** reports
```
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: fixed

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 11]
```diff
@@ -414,6 +430,62 @@ https://api.livechatinc.com/v3.4/reports/chats/ranking \
...
+
+Shows for how long an agent, group, or the whole account was available for chatting during a specified period of time.
+
+#### Specifics
+
+|                     |                                                                    |
+| ------------------- | -------------------------------------------------------------------|
+| **Method URL**      | `https://api.livechatinc.com/v3.4/reports/agents/availability`     |
+| **HTTP method**     | POST, GET                                                          |
+| **Required scopes** | -                                                                  |
```
Comment (janowsiany) [inline src/pages/data-reporting/reports-api/v3.4/index.mdx:448]: it requires some scope, isn't it?

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: src/pages/data-reporting/reports-api/v3.4/index.mdx [hunk 12]
```diff
@@ -88,7 +104,7 @@ List of default properties can be found [**here**](/messaging/references/default
 
 **2)** `<string_filter_type>` can take the following values:
 
-- `exists` (`bool`)
+- `exists` (`bool`) - 💡 available only for the Chats parameter value
```
Comment (oliwiapolec) [inline src/pages/data-reporting/reports-api/v3.4/index.mdx:107]:
```suggestion
- `exists` (`bool`) - 💡 available only for the [Chats](/data-reporting/reports-api/v3.4#for-chats-reports) parameter value
```

File: src/pages/data-reporting/reports-api/v3.4/index.mdx (hunk 11 above)
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: yes, I don't know how I have copied this without it...

Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: added `reports_read` scope

**Review by janowsiany — APPROVED** [review]

File: src/pages/data-reporting/reports-api/v3.4/index.mdx (hunk 9 above)
[the remark this answers is not in the document]
Comment (ale-kacperczyk) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: IMO, this bold **Chats** doesn't look good. It's H3 so it's already in bold

File: src/pages/data-reporting/reports-api/v3.4/index.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (ale-kacperczyk) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: Same issue with bold

**Comment by ale-kacperczyk** [comment]
Just these two remarks about bold text in headlines

File: src/pages/data-reporting/reports-api/v3.4/index.mdx (hunk 9 above)
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: removed bold

File: src/pages/data-reporting/reports-api/v3.4/index.mdx (hunk 10 above)
[the remark this answers is not in the document]
Comment (MaciejSzaflik) [reply src/pages/data-reporting/reports-api/v3.4/index.mdx]: removed bold

**Review by ale-kacperczyk — APPROVED** [review]

**Pull request merged**